app/merchant/views.py

```diff
@@ -14,33 +14,76 @@ def index():
 
 
 @merchant.route('/view/all')
+@merchant.route('/view/all/<int:page>')
+@merchant.route('/view/all/<int:page>/<string:requestlink>')
 @login_required
 @merchant_required
-def listing_view_all():
+def listing_view_all(page=1, requestlink=""):
     """Search for listings"""
-    main_search_term = request.args.get('main-search', "", type=str)
+    main_search_term = request.args.get('mainsearch', "", type=str)
     favorite = True if request.args.get('favorite') == "on" else False
+    sortby = request.args.get('sortby', "", type=str)
     name_search_term = request.args.get('name-search', "", type=str)
     min_price = request.args.get('min-price', "", type=float)
     max_price = request.args.get('max-price', "", type=float)
-    listings = Listing.search(available=True,
-                              favorite=favorite,
-                              min_price=min_price,
-                              max_price=max_price,
-                              name_search_term=name_search_term,
-                              main_search_term=main_search_term)
-
+    listings_raw = Listing.search(available=True,
+                                  favorite=favorite,
+                                  sortby=sortby,
+                                  min_price=min_price,
+                                  max_price=max_price,
+                                  name_search_term=name_search_term,
+                                  main_search_term=main_search_term)
+    listings_paginated = listings_raw.paginate(page, 20, False)
     return render_template('merchant/view_listings.html',
-                           listings=listings,
+                           listings=listings_paginated,
                            main_search_term=main_search_term,
                            min_price=min_price,
                            max_price=max_price,
+                           sortby=sortby,
                            name_search_term=name_search_term,
                            favorite=favorite,
                            cart_listings=current_user.get_cart_listings(),
                            header="All listings")
 
 
+@merchant.route('/cart-action', methods=['POST'])
+@login_required
+@merchant_required
+def cart_action():
```

Comment:
`cart-action` was removed in master, since it's no longer being used. You can delete the entire function. I'm also not sure why there are a couple of regressions from master in your branch